src/lib.rs

```diff
@@ -1311,6 +1312,118 @@ where
     Some(Ratio::new(n1, d1))
 }
 
+impl<T: Clone + Integer + Signed + ToPrimitive + ToBigInt> Ratio<T> {
+    /// Converts the ratio to an `f64`.
+    ///
+    /// The value is guaranteed to be the closest `f64` approximating the ratio. This always
+    /// returns a value, unless the conversion to BigInt for the numerator or denominator fails.
+    ///
+    /// ```
+    /// use num_rational::Rational64;
+    /// assert_eq!(0.5f64, Rational64::new(1, 2).to_f64().unwrap());
+    /// ```
+    pub fn to_f64(&self) -> Option<f64> {
+        assert_eq!(
+            std::f64::RADIX,
+            2,
+            "only floating point implementations with radix 2 are currently supported"
+        );
+        assert!(
+            std::mem::size_of::<isize>() >= 4,
+            "platforms 16 bits and under are not supported"
+        );
+
+        // The following casts all work on 32-bit+ platforms, but may overflow on 16-bit ones.
+        const F64_MAX_EXP: isize = std::f64::MAX_EXP as isize;
+        const F64_MIN_EXP: isize = std::f64::MIN_EXP as isize;
+        const F64_MANTISSA_DIGITS: isize = std::f64::MANTISSA_DIGITS as isize;
```

Comment:
If you change these declarations to i32 or i64, could you avoid the lack of 16-bit support?

Reply:
I think so; this would mean that we wouldn't support BigInts with more than 2^32 or 2^64 bits, which seems reasonable enough (`BigInt::bits()` currently returns an `usize`, which is the main reason for these casts).

Reply:
I think you should change these casts to usize, so that you have the full range of values for later and allow 16-bit support.

See my other comment for why you should change them to usize. 

As for why you should allow 16-bit architectures, the max f64 is about [1.8e+308](https://stackoverflow.com/questions/26473956/maximum-possible-value-for-f64), while the biggest `BigUint` that can be represented with `2^16-1` bits is [`1e+19728`](https://www.wolframalpha.com/input/?i=2%5E((2%5E16)-1)+-+1). These are completely different levels of big. We should be able to fail-early (i.e. return infinity) even on 16-bit platforms.